### PR TITLE
Add Dockerfile for NGINX Open Source for Openshift

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes

This PR adds a new Dockerfile for NGINX Ingress Controller for Openshift for NGINX open source.

### CODE REVIEW

File: build/openshift/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	&& useradd --system --gid nginx --no-create-home --home-dir /nonexistent --comment "nginx user" --shell /bin/false --uid 101 nginx \
+	&& echo "[nginx]" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "name=nginx repo" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "baseurl=https://nginx.org/packages/mainline/rhel/8/\$basearch/" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "gpgcheck=0" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "enabled=1" >> /etc/yum.repos.d/nginx.repo \
+	&& echo "module_hotfixes=true" >> /etc/yum.repos.d/nginx.repo \
+	&& yum update -y \
+	&& yum install -y nginx-${NGINX_VERSION} \
+	&& nginx \
```
Comment (pleshakov) [inline build/openshift/Dockerfile]: why `nginx`? 🤔  we're installing ` nginx-${NGINX_VERSION}` in the previous line

File: build/openshift/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	&& mkdir -p /var/lib/nginx \
+	&& mkdir -p /etc/nginx/secrets \
+	&& setcap 'cap_net_bind_service=+ep' /usr/sbin/nginx \
+	&& setcap 'cap_net_bind_service=+ep' /usr/sbin/nginx-debug \
+	&& chown -R nginx:0 /etc/nginx \
+	&& chown -R nginx:0 /var/cache/nginx \
+	&& chown -R nginx:0 /var/lib/nginx \
+	&& rm /etc/nginx/conf.d/*
```
Comment (pleshakov) [inline build/openshift/Dockerfile]: In the plus image, you removing the ".repo" file https://github.com/nginxinc/kubernetes-ingress/blob/master/build/openshift/DockerfileForPlus#L48
Why not here?

File: build/openshift/Dockerfile (hunk 1 above)
Comment (Rulox) [reply build/openshift/Dockerfile]: This is not installing, this is starting nginx binary for the first time. I followed the official docs for RHEL from [here](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/#installing-a-prebuilt-centos-rhel-package-from-the-official-nginx-repository).

**Comment by Rulox** [comment]
I had to update the Dockerfiles with Labels so we can get the RH Certification for Openshift. Please have another look. cc @pleshakov @Dean-Coakley @lucacome  Thanks

**Comment by Rulox** [comment]
Merging, labels can be updated if needed.

**Pull request merged**